src/Esprima/JavascriptParser.cs

```diff
@@ -507,7 +509,40 @@ private void ExpectKeyword(string keyword)
     [MethodImpl(MethodImplOptions.AggressiveInlining)]
     private protected bool Match(string value)
     {
-        return _lookahead.Type == TokenType.Punctuator && value.Equals((string) _lookahead.Value!);
+        // ReSharper disable once InlineTemporaryVariable
+        ref readonly var token = ref _lookahead;
+        return token.Type == TokenType.Punctuator && value.Equals((string) token.Value!);
+    }
+
+    /// <summary>
+    /// Return true if the next token matches the specified punctuator and consumes the next token.
+    /// </summary>
+    [MethodImpl(MethodImplOptions.AggressiveInlining)]
+    private protected bool ConsumeMatch(string value)
+    {
+        if (Match(value))
+        {
+            NextToken();
+            return true;
+        }
+
+        return false;
+    }
+
+    /// <summary>
+    /// Return true if the next token matches the specified punctuator.
+    /// </summary>
+    private bool Match(char value1, char value2, char value3, char value4)
```

Comment:
nit: a name like `MatchAny` would describe the purpose better

Reply:
changed name